### PR TITLE
publishing-bot: add job to run go tests

### PR DESCRIPTION
**Pull request by cpanato**
Fixes: https://github.com/kubernetes/publishing-bot/issues/247

part of: https://github.com/kubernetes/publishing-bot/pull/249

/assign @nikhita

### CODE REVIEW

File: config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml [hunk 1]
```diff
@@ -12,6 +12,18 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-release-publishing-bot
       testgrid-tab-name: build
+  - name: pull-publishing-bot-test
+    always_run: true
+    decorate: true
+    path_alias: k8s.io/publishing-bot
+    spec:
+      containers:
+      - image: golang:1.13
```
Comment (nikhita) [inline config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml:21]: I think we should be able to update this to 1.16 for publishing-bot (all prowjobs)

Comment (nikhita) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml:21]: ah, just saw https://github.com/kubernetes/publishing-bot/issues/248 :smile:

Comment (cpanato) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml:21]: should we do it in a follow-up or in this PR?

Comment (nikhita) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml:21]: should we be fine to do it in this PR IMO

Comment (cpanato) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml:21]: will do thanks!

Comment (cpanato) [reply config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml]: update to go1.16, ptal @nikhita

**Review by nikhita — APPROVED** [review]
Thank you!! :pray: 

/lgtm
/approve

**Pull request merged**